src/xdp/extensionset.c

```diff
@@ -326,15 +326,31 @@ XdpExtensionSetCreate(
         goto Exit;
     }
 
-    Set->Type = Type;
-    Set->Count = ReservedExtensionCount;
+    Status = XdpExtensionSetInitialize(
+        Type, ReservedExtensions, ReservedExtensionCount, Set);
 
-    for (UINT16 Index = 0; Index < Set->Count; Index++) {
-        XDP_EXTENSION_ENTRY *Entry = &Set->Entries[Index];
+Exit:
+
+    return Status;
+}
+
+NTSTATUS
+XdpExtensionSetInitialize(
+    _In_ XDP_EXTENSION_TYPE Type,
+    _In_opt_count_(ReservedExtensionCount) const XDP_EXTENSION_REGISTRATION *ReservedExtensions,
+    _In_ UINT16 ReservedExtensionCount,
+    _Inout_ XDP_EXTENSION_SET *ExtensionSet
+    )
+{
+    ExtensionSet->Type = Type;
+    ExtensionSet->Count = ReservedExtensionCount;
```

Comment:
We should assert `ExtensionSet->Count >= ReservedExtensionCount` to make sure the input ExtensionSet is large enough for the current parameters.

Reply:
We'd also need to set the `Count` within `XdpExtensionSetCreate`

Reply:
will queue up a fix